src/webgpu/util/copy_to_texture.ts

```diff
@@ -165,31 +227,88 @@ export class CopyToTextureUtils extends GPUTest {
 
   // MAINTENANCE_TODO(crbug.com/dawn/868): Should be possible to consolidate this along with texture checking
   checkBufferWithRowPitch(
-    actual: Uint8Array,
-    exp: Uint8Array,
+    actual: TypedArrayBufferView,
+    expected: TypedArrayBufferView,
     width: number,
     height: number,
     rowPitch: number,
     bytesPerPixel: number,
-    isFp16: boolean
+    dstFormat: RegularTextureFormat
   ): string | undefined {
     const bytesPerRow = width * bytesPerPixel;
+
     // When dst format is fp16 formats, the expectation and real result always has 1 bit difference in the ending
     // (e.g. CC vs CD) if there needs some alpha ops (if alpha channel is not 0.0 or 1.0). Suspect it is errors when
     // doing encoding. We check fp16 dst texture format with 1-bit ULP tolerance.
-    if (isFp16) {
+    if (isFp16Format(dstFormat)) {
+      const exp = new Uint16Array(
+        expected.buffer,
+        expected.byteOffset / Uint16Array.BYTES_PER_ELEMENT,
+        expected.byteLength / Uint16Array.BYTES_PER_ELEMENT
+      );
+      const check = new Uint16Array(
+        actual.buffer,
+        actual.byteOffset / Uint16Array.BYTES_PER_ELEMENT,
+        actual.byteLength / Uint16Array.BYTES_PER_ELEMENT
+      );
+
       for (let y = 0; y < height; ++y) {
-        const expRow = exp.subarray(y * bytesPerRow, bytesPerRow);
-        const checkResult = checkElementsBetween(actual.subarray(y * rowPitch, bytesPerRow), [
-          i => (expRow[i] > 0 ? expRow[i] - 1 : expRow[i]),
-          i => expRow[i] + 1,
+        const expRow = exp.subarray(
+          (y * bytesPerRow) / exp.BYTES_PER_ELEMENT,
+          bytesPerRow / exp.BYTES_PER_ELEMENT
+        );
+        const checkRow = check.subarray(
+          (y * bytesPerRow) / check.BYTES_PER_ELEMENT,
+          bytesPerRow / check.BYTES_PER_ELEMENT
+        );
+        const expRowF32 = new Float32Array(expRow.length);
+        const checkRowF32 = new Float32Array(checkRow.length);
+        checkRow.forEach((v: number, i: number) => {
+          checkRowF32[i] = float16BitsToFloat32(v);
+        });
+        expRow.forEach((v: number, i: number) => {
+          expRowF32[i] = float16BitsToFloat32(v);
+        });
+        const checkResult = checkElementsBetween(checkRowF32, [
+          i => expRowF32[i] - 0.001,
+          i => expRow[i] + 0.001,
```

Comment:
```suggestion
          i => expRowF32[i] + 0.001,
```
I think. Perhaps worth renaming `expRow` to `expRowF16Bits` so this error would be more apparent.

Reply:
Update the name and fix this bug. Thanks!